src/PDPVerifier.sol

```diff
@@ -316,16 +325,19 @@ contract PDPVerifier is Initializable, UUPSUpgradeable, OwnableUpgradeable {
         require(rootData.length > 0, "Must add at least one root");
         require(proofSetOwner[setId] == msg.sender, "Only the owner can add roots");
         uint256 firstAdded = nextRootId[setId];
+        uint256[] memory rootIds = new uint256[](rootData.length);
+
 
         for (uint256 i = 0; i < rootData.length; i++) {
             addOneRoot(setId, i, rootData[i].root, rootData[i].rawSize);
+            rootIds[i] = firstAdded + i;
         }
+        emit RootsAdded(setId, rootIds);
```

Comment:
Comparing this event to the PDPListener event emitted below:

```solidity
PDPListener(listenerAddr).rootsAdded(setId, firstAdded, rootData, extraData);
```

I see PDPListener is also receiving `extraData`.

What is the purpose of that extra data? Would it make sense to include it in the `RootsAdded` event too?

```solidity
emit RootsAdded(setId, rootIds, extraData);
```

I don't see any obvious use case for that extra data in Spark; this may be a YAGNI thing. I am mentioning it here for your consideration.

Reply:
@bajtos That `extraData` is for an end user such as Storacha to get "some" data as a callback to it's listener/service (auth etc.) that it negotiated offchain with the client. I don't think we need it in events for now.

It's also not currently obvious if the services/listeners themselves will need this extra data.

Reply:
Its safe to assume that extra data reporting can be handled by the listener.

Reply:
Makes sense, thanks for the explanation 👍🏻